### PR TITLE
Document `Radical`

### PR DESCRIPTION
**Pull request by rcurtin**
I documented the RADICAL algorithm for ICA.  This one actually turned out to be very easy to do.  Changes:

 * Deprecated `Radical::DoRadical()` in favor of the new `Radical::Apply()`, which matches classes like `PCA` and other transformations better.
 * Templatized `Radical::Apply()` so users can use dense floating-point matrix types.
 * Adapted tests to use other template types.
 * Moved auxiliary memory class members into the `Radical::Apply()` function directly.
 * Allowed automatic estimation of `m` parameter (like `radical_main.cpp` does).
 * Updated `SEE_ALSO()` links for all the bindings I could find to point to Markdown files instead of C++ source.

### CODE REVIEW

File: src/mlpack/methods/radical/radical.hpp [hunk 1]
```diff
@@ -71,29 +71,55 @@ class Radical
    * @param matW Estimated unmixing matrix, where matY = matW * matX.
    * @param timers Optional Timers struct for storing timing information.
    */
-  void DoRadical(const arma::mat& matX,
-                 arma::mat& matY,
-                 arma::mat& matW,
+  template<typename MatType = arma::mat>
+  [[deprecated("Will be removed in mlpack 5.0.0.  Use Apply() instead.")]]
+  void DoRadical(const MatType& matX,
+                 MatType& matY,
```
Comment (shrit) [inline src/mlpack/methods/radical/radical.hpp:77]: This should have been really old,

File: src/mlpack/methods/radical/radical.hpp [hunk 2]
```diff
@@ -71,29 +71,55 @@ class Radical
...
   //! Two-dimensional version of RADICAL.
-  double DoRadical2D(const arma::mat& matX,
-                     util::Timers& timers = IO::GetTimers());
+  template<typename MatType>
+  typename MatType::elem_type DoRadical2D(
+      const MatType& matX,
+      const size_t m,
+      MatType& perturbed, // auxiliary memory
+      MatType& candidate, // auxiliary memory
+      util::Timers& timers = IO::GetTimers());
```
Comment (shrit) [inline src/mlpack/methods/radical/radical.hpp:122]: Can we deprecate this one in favour of Apply2D? So we have an API compatibility, especially when we are in mlpack 5 no one will understand why we have Apply and DoRadical2D

**Review by shrit — COMMENTED** [review]
Looks, good it was easier to review compared to what I would have thought, I just left a couple of comments

File: src/mlpack/methods/radical/radical.hpp (hunk 2 above)
Comment (rcurtin) [reply src/mlpack/methods/radical/radical.hpp]: You're right, that is a more consistent name.  I don't consider `DoRadical2D()` a "public API" function, so it's ok to change.  (For this reason I didn't document it.)  So I just changed `DoRadical2D()` to `Apply2D()` in 6a2e9cb33d. :+1:

**Review by shrit — APPROVED** [review]
looks good to me, the same things goes for github actions

**Comment by rcurtin** [comment]
The Windows build looks to be a spurious failure unrelated to RADICAL, so I will go ahead and merge. :rocket:

**Pull request merged**